OpenEdXMobile/src/main/java/org/edx/mobile/player/PlayerFragment.java

```diff
@@ -1707,35 +1710,54 @@ private void hideCCPopUp(){
     /**
      * This function is used to display CC data when a transcript is selected
      */
-    private void displaySrtData(){
-        try{
-            if (subtitleDisplayHandler != null) {
-                subtitleDisplayHandler.removeCallbacks(subtitleProcessesor);
-            }
-            resetClosedCaptioning();
-            if(srtList!=null && srtList.size()>0){
-                final String languageSubtitle = getSubtitleLanguage();
-                if(languageSubtitle!=null){
-                    srt = srtList.get(languageSubtitle);
-                    if (srt != null) {
-                        try{
-                            if(player!=null){
-                                environment.getAnalyticsRegistry().trackShowTranscript(videoEntry.videoId,
-                                        player.getCurrentPosition()/AppConstants.MILLISECONDS_PER_SECOND,
-                                        videoEntry.eid, videoEntry.lmsUrl);
+    private void displaySrtData() {
+        if (subtitleDisplayHandler != null) {
+            subtitleDisplayHandler.removeCallbacks(subtitleProcessesor);
+        }
+        resetClosedCaptioning();
+        if (srtList != null && srtList.size() > 0) {
+            String languageSubtitle = getSubtitleLanguage();
+
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
+                if (languageSubtitle == null) {
+                    // Check if captioning is enabled in accessibility settings and set the captioning language implicitly
+                    final CaptioningManager cManager = (CaptioningManager) getContext().getSystemService(Context.CAPTIONING_SERVICE);
+                    if (cManager.isEnabled()) {
+                        final String defaultCcLanguage;
+                        {
```

Comment:
@miankhalid  I don't think so, there is any need of an extra block

`cManagerLocale` is already have a limited scope with in 
`if (cManager.isEnabled()) {` block